src/main/scala/com/oysterbooks/scavro/plugin/AvroCodegenPlugin.scala

```diff
@@ -10,10 +10,15 @@ object AvroCodegenPlugin extends AutoPlugin {
     val avroSchemaFiles = settingKey[Seq[File]]("Avro schema files to compile")
     val avroProtocolFiles = settingKey[Seq[File]]("Avro Protocol files to compile")
     val avroIDLFiles = settingKey[Seq[File]]("Avro IDL files to compile")
+    val avroDataFiles = settingKey[Seq[File]]("Avro datafiles to compile")
 
     val showAvroCompilerOutput = settingKey[Boolean]("Hides messages from avro compiler")
 
     val avroCodeOutputDirectory = settingKey[File]("Location where generated Java code will be placed")
+    val scalaCodeOutputDirectory = settingKey[File]("Location where generated Scala wrapper code will be placed")
+
+    val scalaCustomTypes = settingKey[Map[String, String]]("Customize Avro to Scala type map by type")
+    val scalaCustomNamespace = settingKey[Map[String, String]]("Custom namespace of generated Scala wrapper code")
```

Comment:
There is a small section of documentation [here](https://github.com/julianpeeters/scavro/blob/code_generation/README.md#scavro-plugin), but reading it now I recognize it as blurry and hard to read. I think I can edit to make it more clear and concise. 

Currently the default is to append "model" to the preexisting avro namespace (so the `com.oysterbooks.scavrodemo.idl` becomes `com.oysterbooks.scavrodemo.idl.model`). I prefer the demo's `com.oysterbooks.scavrodemo.model`, but some namespaces won't survive being truncated. Maybe a small ruleset could check for a terminal `.idl` and replace it with `model` would help? Seems to me like just more complication for a user to have to reason about, but of course I'm open to suggestions. 

Your point about the naming is well taken. Any preference on what makes a name too long?  `avroScalaCodeOutputDirectory` better?